File: POO - Python & Java/practica-1/src/uiMain/funcionalidades/Urgencia.java
package uiMain.funcionalidades;

import java.util.Random;
import java.util.Scanner;
import java.util.HashMap;

import uiMain.Utilities;

import gestorAplicacion.Hospital;
import gestorAplicacion.Quirofano;
import gestorAplicacion.Paciente;
import gestorAplicacion.Doctor;
import gestorAplicacion.Sintomas;
import gestorAplicacion.Factura;

public class Urgencia {
	
	/*
	 * nuevaUrgencia: Crea una nueva instancia Paciente
	 * PD: Este es el metodo que se ejecuta desde interfaz
	 */
	public static void nuevaUrgencia(Hospital hospital, Scanner in) {
		Random rand = new Random();
		
		// DETERMINAR ALEATORIAMENTE EN QUE LLEGA EL PACIENTE
		String[] transporteOpciones = new String[]{"helicoptero", "ambulancia", "carro particular"};
		String transporte = transporteOpciones[rand.nextInt(transporteOpciones.length)];
		System.out.println("Ha llegado alguien en " + transporte);
		
		Utilities.waitInput(in);
		
		// SE LE DETERMINA ALEATORIAMENTE UNA SERIE DE SINTOMAS Y DE ACUERDO A ESO UNA TRAMA
		HashMap<String, Sintomas> opciones = new HashMap<String, Sintomas>(){{
			put("le dio un paro cardiaco mientras se comia una hamburguesa", Sintomas.DOLORPECHO);
			put("le envenenaron la comida", Sintomas.DOLORESTOMAGO);
			put("tuvo un shock electrico y le dio un paro cardiaco", Sintomas.DEFORMACION);
			put("le dio un paro cardia", Sintomas.DOLORPECHO);
			put("se puso un stricker y ahora la piel se ve bastante mal", Sintomas.PIELRARA);
			put("lamio la calle y ahora tiene una tos increible", Sintomas.TOS);
		}};
		String[] causasOpciones = opciones.keySet().toArray(new String[2]);
		String causa = causasOpciones[rand.nextInt(causasOpciones.length)];
		System.out.println("El paciente llego a urgencias porque " + causa);
		Sintomas sintoma = opciones.get(causa);

		// DETERMINAR 
		boolean identificado = rand.nextBoolean();
		String nombre;
		if(identificado) {
			
			// DETERMINAR ALEATORIAMENTE DE UNA LISTA DE OPCIONES, EL NOMBRE DEL PACIENTE
			String[] nombreOpciones = new String[]{"Juan", "Pedro", "Camilo",
					"Julian", "Daniel", "Sebastian", "Maria", "Juliana", "Natalia"};
			nombre = nombreOpciones[rand.nextInt(nombreOpciones.length)];
			
			int cedula = rand.nextInt(1000000,19999999); // DA ALEATORIAMENTE UNA CEDULA
			
			System.out.println("El paciente se llama " + nombre + " y esta identificado con"
					+ " cedula de ciudadania " + cedula);
			
			
			
			
			// SE DETERMINA SI EL PACIENTE ES MILITAR, DOCTOR O OTRO
			String[] profesionesOpciones = new String[]{"militar", "doctor", "otro"};
			String profesion = profesionesOpciones[rand.nextInt(profesionesOpciones.length)];
			
			// DEPENDIENDO DE SU PROFESION SE LE APLICA DESCUENTO
			double descuento;
			switch (profesion) {
			case "doctor":
				System.out.println("El paciente es un doctor y tiene un descuento del 15%");
				descuento = 0.15;
				break;
			case "militar":
				System.out.println("El paciente es militar y redimio el codigo MILITAR, que le da un 50% de descuento en su cirugia B)");
				descuento = 0.50;
				break;
			default:
				descuento = 0.0;
				break;
			}
			
			Utilities.waitInput(in);
			
			// SE CREA UNA INSTANCIA DEL PACIENTE PARA REFERENCIAR
			Paciente paciente = new Paciente(nombre, cedula, sintoma);
			
			// SE BUSCA UN QUIROFANO URGENTE
			Quirofano quirofano = hospital.apartarQuirofanoUrgencia(paciente);
			
			// SI NO SE CONSIGUIO
			if(quirofano == null) {
				System.out.println("F, el paciente murio porque no habia ciruganos/doctores disponibles X_X");
			} else {
				System.out.println("Ya lo estan operando");
				Factura factura = new Factura(sintoma.getEspecialidad().getCostoCirujia(), descuento);	// GENERAR FACTURA
				System.out.println("La factura ha llegado por " + factura.getValorPagar() + " la deuda ha sido generada y se espera el pago");
			}
		} else {
			System.out.println("El paciente no esta identificado :(, esto genera problemas");
			Doctor doctor = hospital.getcirujanoEspecialidad(sintoma.getEspecialidad());
			
			System.out.println("El doctor " +  doctor + " justo esta pasando y es muy buena gente");
			System.out.println("Segun las politicas del hospital, no podemos operar al paciente al no estar identificado");
			System.out.println("Pero... podriamos operarlo de todas maneras  ( ͡° ͜ʖ ͡°)");
			System.out.println("Puedes decir por el doctor y decidir si operar al paciente,"
					+ "\nINGRESA ;) para operar, de lo contrario, entedere tu señal como que no quieres operar al paciente\n");
			
			in.nextLine();
			String election = in.nextLine();
			if(election.equals(";)")) {
				// MUESTRA LOS QUIROFANOS QUE HAY EN EL HOSPITAL
				mostrarQuirofanos(hospital);
				System.out.println("Ingresa el numero del quirofano al que quieres entrar");
				int electionQuirofano = in.nextInt();
				boolean suceed = hospital.apartarQuirofano(electionQuirofano, null, doctor); // SI EL QUIROFANO ESTA DISPONIBLE RETORNA TRUE
				
				// DEPENDIENDO DE SI SE HIZO UNA BUENA ELECCION O NO (SI EL QUIROFANO QUE SE ELIJIO ESTABA DISPONIBLE)
				if(!suceed) { 
					System.out.println("Has entrado a donde estaba otro doctor y te ha sapeado :(, han despedido al doctor " + doctor);
				} else {
					System.out.println("Has elejido sabiamente, ya estan operando al paciente");
				}
			} 
			// SI NO SE QUISO ARRIESGAR A OPERAR AL PACIENTE INDOCUMENTADO
			else {
				System.out.println("El paciente ha muerto");
			}
		}
		
		System.out.println("La emergencia ha finalizado, volviendo a menu");
		Utilities.waitInput(in);
	}
	
	public static void mostrarQuirofanos(Hospital hospital) {
		System.out.println("Vamos a conseguir ");
		int  i = 0;
		for (Quirofano quirofano : hospital.getQuirofanos()) {
			System.out.println("[" + i + "] " + quirofano);
			i++;
		}
	}
}

File: POO - Python & Java/practica-2/uiMain/funcionalidades/urgencia.py

#
# IMPORTS
#
from tkinter import messagebox
from gestorAplicacion.Especialidad import Especialidad

from gestorAplicacion.Sintomas import Sintomas
from gestorAplicacion.Paciente import Paciente
from gestorAplicacion.Hospital import Hospital
from gestorAplicacion.Quirofano import Quirofano
from gestorAplicacion.Doctor import Doctor
from gestorAplicacion.Factura import Factura

import random




#
# Main Class
# 


class Urgencia:
    """
    @description: Un paciente llega a emergencias, quien sera? eso lo determinara el azar
    """
    def nueva_urgencia(hospital):
        transporte_opciones = ["helicoptero", "ambulancia", "carro particular"]
        transporte = random.choice(transporte_opciones)
        messagebox.showinfo(message=f"Ha llegado alguien en {transporte}")
        
        opciones = {
            "le dio un paro cardiaco mientras se comia una hamburguesa": Sintomas.DOLORPECHO,
            "le envenenaron la comida": Sintomas.DOLORESTOMAGO,
            "tuvo un shock electrico y le dio un paro cardiaco": Sintomas.DEFORMACION,
            "le dio un paro cardia": Sintomas.DOLORPECHO,
            "se puso un stricker y ahora la piel se ve bastante mal": Sintomas.PIELRARA,
            "lamio la calle y ahora tiene una tos increible": Sintomas.TOS
        }
        
        causa = random.choice(list(opciones.keys()))
        messagebox.showinfo(message=f"El paciente llego a urgencias porque {causa}")
        
        identificado = random.choice([True, False])
        
        output_final = ""
        
        if identificado:
            nombre_opciones = ["Juan", "Pedro", "Camilo", "Julian", "Daniel", "Sebastian", "Maria", "Juliana", "Natalia"]
            nombre = random.choice(nombre_opciones)
            
            cedula = random.randint(1000000, 19999999)
            
            messagebox.showinfo(message=f"El paciente se llama {nombre} y esta identificado con la cedula {cedula}")
            
            profesiones_opciones = ["militar", "doctor", "otro"]
            profesion = random.choice(profesiones_opciones)
            
            descuento = 0
            if profesion == "militar":
                messagebox.showinfo(message=f"El paciente es militar y redimio el codigo MILITAR, que le da un 50% de descuento en su cirugia B)")
                descuento = 0.50
            elif profesion == "doctor":
                messagebox.showinfo(message=f"El paciente es un doctor y tiene un descuento del 15%")
                descuento = 0.15
            
            # CREAR INSTANCIA PACIENTE
            paciente = Paciente(nombre, cedula, random.choice([True, False]), [opciones[causa]], -1)
            quirofano_urgente = hospital.apartarQuirofanoUrgencia(paciente)
            
            if quirofano_urgente == None:
                output_final = "F, el paciente murio porque no habia ciruganos/doctores disponibles X_X"
            else:
                factura = Factura(opciones[causa].getEspecialidad().costoCirujia, descuento)
                output_final = f"Ya lo estan operando\nLa factura llego por {factura.getValorPagar()}"
        else:
            messagebox.showwarning(message="El paciente no esta identificado :(, esto genera problemas")
            doctor = hospital.getcirujanoEspecialidad(opciones[causa])
            
            if doctor:
                messagebox.showwarning(message="Segun las politicas del hospital, no podemos operar al paciente al no estar identificado")
                messagebox.showinfo(message="Pero... podriamos operarlo de todas maneras  ( ͡° ͜ʖ ͡°)")
                operar = messagebox.askyesno(message="Quieres operar el paciente a escondidas?")
                if operar:
                    output_final = "Estan operando al paciente"
                else:
                    output_final = "El paciente murio X_X"
            else:
                doctor = hospital.getcirujanoEspecialidad(Especialidad.GENERAL)
                if doctor:
                    output_final = f"El doctor {doctor.getNombre()} va a realizar la operacion, es de medicina general, sirve"
                else:
                    output_final = "Igual no habia doctores, no hubo nada que hacer, el paciente murio X_X"
            
        return output_final